AsyncPoco/Core/PocoData.cs

```diff
@@ -358,10 +358,10 @@ private static void AddConverterToStack(ILGenerator il, Func<object, object> con
 				{
 					return src => EnumMapper.EnumFromString(dstType, (string)src);
 				}
-				else
-				{
-					return src => Convert.ChangeType(src, dstType, null);
-				}
+				return src => {
+					var actualDstType = Nullable.GetUnderlyingType(dstType) ?? dstType;
```

Comment:
Adding this line to GetUnderlyingType was vital in getting the Tests to work for SQLite. I checked my approach against NPoco and PetaPoco and they too added this to their code base (albeit in a slightly different manner).